.github/workflows/ci_cd.yml

```diff
@@ -417,7 +418,34 @@ jobs:
         uses: ./.github/actions/load-img
         with:
           run_id: ${{ github.run_id }}
-          setup_images: upstream_db ingestion_server api=api_dev
+          setup_images: upstream_db ingestion_server
+
+      # Sets build args specifying versions needed to build Docker image.
+      - name: Prepare build args
+        id: prepare-build-args
+        run: |
+          just versions | tee "$GITHUB_OUTPUT"
+
+      - name: Setup Docker Buildx
+        uses: docker/setup-buildx-action@v3
+        with:
+          install: true
+
+      - name: Build API dev image
+        uses: docker/build-push-action@v5
+        with:
+          context: api
+          target: api
+          push: false
+          load: true
+          tags: openverse-api
+          cache-from: type=gha,scope=api_dev
```

Comment:
I'm not 100% sure how this works, but if we cache on `scope=api`, would that allow us to reuse the layers from the existing `api` build, if it's there? :thinking: 

A small improvement if it's even possible, so no worries if it isn't or if doing that is very complex.

Reply:
I was worried that layers from `api-dev` that differ from those in the `api` (especially after the difference of `--dev`/`--prod` in the PDM install step) would overwrite the cache and vice versa making both image builds slower by introducing cache misses.

Anyway, this separate cache automatically speeds up the builds across CI runs so it's still very fast.